lib/grp.gi

```diff
@@ -130,14 +130,103 @@ end);
 InstallOtherMethod(MinimalGeneratingSet,"fallback method to inform user",true,
   [IsObject],0,
 function(G)
+  local
+    check,GbyGk,
+    Gkm1byGk,
+    Gkm1byGk_L,
+    Gkm1byGk_elem_reps,
+    Gkm1byGk_gen,
+    Gkm1byGk_gen_reps,
+    mingenset_km1_reps,
+    mingenset_k_reps,
+    temp,i,j,l,L,x,xl,y,prev,gmod,N,g,g0,g1,s,r,stop,
+    cs,phi_GbyG1,GbyG1,phi_GbyGk,phi_Gkm1byGk,k;
   if IsGroup(G) and IsSolvableGroup(G) then
     TryNextMethod();
+  elif IsGroup(G) and IsFinite(G) then
+    cs := ChiefSeries(G);
```

Comment:
This fallback method seems to me like the wrong place for this code. I would suggest to put it into its own function, say `BindGlobal("MinimalGeneratingSetUsingChiefSeris", ...);`, further up in this file and then we can call it from any place we deem appropriate

Reply:
Also, since this is based on a paper, there should be an initial comment which states something like "This function implements the algorithm described in PAPER."

Some more helpful comments in general would be good...

Reply:
I made these changes.